ap/calibration/jets.py

```diff
@@ -0,0 +1,432 @@
+# coding: utf-8
+
+"""
+Calibration methods for jets
+"""
+import os
+
+import law
+
+from ap.util import maybe_import, memoize
+from ap.calibration import calibrator
+from ap.columnar_util import set_ak_column
+
+np = maybe_import("numpy")
+ak = maybe_import("awkward")
+coffea = maybe_import("coffea")
+
+coffea_extractor = maybe_import("coffea.lookup_tools.extractor")
+coffea_jetmet_tools = maybe_import("coffea.jetmet_tools")
+coffea_txt_converters = maybe_import("coffea.lookup_tools.txt_converters")
+
+
+#
+# first, some utility functions
+#
+
+@memoize
+def get_lookup_provider(files, conversion_func, provider_cls, names=None):
+    """
+    Create a coffea helper object for looking up information in files of various formats.
+
+    This function reads in the `files` containing lookup tables (e.g. JEC text files), extracts
```

Comment:
For the doc strings it might be better to use e.g. `*files*` to highlight arguments. RsT unfortunately doesn't play nicely with the default markdown style.

Reply:
Good point! I was thinking the (single) backticks will render as monospace in RsT as well, but I guess that is not as reliable as it depends on the style. I'll change them to asterisks for the args, and qualify the backticks with `:py:class:` etc. to conform to RsT syntax.

Reply:
Done